tests/bug_gen/adapters/test_typescript.py

```diff
@@ -0,0 +1,148 @@
+def entities(test_file_ts):
+    entities = []
+    get_entities_from_file_ts(entities, str(test_file_ts))
+    return entities
+
+
+def test_get_entities_from_file_ts_count(entities):
+    """Test that we find a reasonable number of entities in buffer.ts."""
+    # Should find the class and many methods (at least 10)
+    assert len(entities) == 30
+
+
+def test_get_entities_from_file_ts_max(test_file_ts):
+    """Test that max_entities parameter limits the number of results."""
+    entities = []
+    get_entities_from_file_ts(entities, str(test_file_ts), 3)
+    assert len(entities) == 3
+
+
+def test_get_entities_from_file_ts_names(entities):
+    """Test that key method names are present (no class entity expected)."""
+    names = [e.name for e in entities]
+    assert "constructor" in names
+    assert "append" in names
+    assert "get" in names
+    assert "_allocQueue" in names
+    assert "queue" in names
+
+
+def test_get_entities_from_file_ts_line_ranges(entities):
+    """Test that at least one method has a plausible line range."""
+    # At least one method should be small
+    method = next(e for e in entities if e.name == "constructor")
+    assert method.line_end - method.line_start < 20
+
+
+def test_get_entities_from_file_ts_extensions(entities):
+    """Test that all entities have the correct file extension."""
+    assert all(e.ext == "ts" for e in entities)
+
+
+def test_get_entities_from_file_ts_file_paths(entities, test_file_ts):
+    """Test that all entities have the correct file path."""
+    assert all(e.file_path == str(test_file_ts) for e in entities)
+
+
+def test_get_entities_from_file_ts_signatures(entities):
+    """Test that signatures are non-empty and plausible for key entities."""
+    method = next(e for e in entities if e.name == "constructor")
+    assert method.signature.startswith("constructor(")
+    for e in entities:
+        assert e.signature.strip(), f"Entity {e.name} should have a non-empty signature"
+
+
+def test_get_entities_from_file_ts_stubs(entities):
+    """Test that all entities have a non-empty stub."""
+    for e in entities:
+        assert e.stub.strip(), f"Entity {e.name} should have a non-empty stub"
+        assert "TODO" in e.stub
+
+
+def test_get_entities_from_file_ts_complexity(entities):
+    """Test that complexity is always >= 1 for all methods."""
+    for e in entities:
+        assert e.complexity >= 1
+
+
+def test_get_entities_from_file_ts_exact(entities):
+    """Test that the extracted entities match the expected list exactly."""
+    expected = [
+        (
+            "constructor",
+            "constructor(map: SourceMap | null, indentChar: string)",
+            36,
+            45,
+        ),
+        ("_allocQueue", "_allocQueue()", 70, 84),
+        (
+            "_pushQueue",
+            "_pushQueue(\n  char: number,\n  repeat: number,\n  line: number | undefined,\n  column: number | undefined,\n  filename: string | undefined,\n) {\n  const cu",
```

Comment:
It looks like signature includes part of the method body for `_pushQueue` (and other methods that have multi-line signatures).

Reply:
Oh yeah, this also looks sort of parsed incorrectly... Will simplify `def signature`, it's a little complicated now that I'm looking at it.